diff -Nur heimdal-1.2.1-orig/cf/check-var.m4 heimdal-1.2.1/cf/check-var.m4
--- heimdal-1.2.1-orig/cf/check-var.m4	2008-07-28 14:04:49.000000000 +0200
+++ heimdal-1.2.1/cf/check-var.m4	2009-09-18 18:12:45.000000000 +0200
@@ -19,7 +19,7 @@
 if test "$ac_foo" = yes; then
 	AC_DEFINE_UNQUOTED(AS_TR_CPP(HAVE_[]$1), 1, 
 		[Define if you have the `]$1[' variable.])
-	m4_ifval([$2], AC_CHECK_DECLS([$1],[],[],[$2]))
+	m4_ifval([$2], AC_CHECK_DECLS([$1],[],[],[[$2]]))
 fi
 ])
 
diff -Nur heimdal-1.2.1-orig/cf/roken-frag.m4 heimdal-1.2.1/cf/roken-frag.m4
--- heimdal-1.2.1-orig/cf/roken-frag.m4	2008-07-28 14:04:49.000000000 +0200
+++ heimdal-1.2.1/cf/roken-frag.m4	2009-09-18 18:12:26.000000000 +0200
@@ -586,11 +586,11 @@
 #include <err.h>
 #endif])
 
-AC_CHECK_DECLS([optarg, optind, opterr, optopt, environ],[],[][
+AC_CHECK_DECLS([optarg, optind, opterr, optopt, environ],[],[],[[
 #include <stdlib.h>
 #ifdef HAVE_UNISTD_H
 #include <unistd.h>
-#endif])
+#endif]])
 
 dnl
 dnl Check for fields in struct tm
