--- heimdal-0.8.1/lib/hdb/Makefile.am.orig	2007-04-23 18:23:48.000000000 +0200
+++ heimdal-0.8.1/lib/hdb/Makefile.am	2007-07-05 20:57:31.063110405 +0200
@@ -39,11 +39,13 @@
 
 ldap_so = hdb_ldap.la
 hdb_ldap_la_SOURCES = hdb-ldap.c
-hdb_ldap_la_LDFLAGS = -module
+hdb_ldap_la_LDFLAGS = -module -avoid-version
+hdb_ldap_la_LIBADD = $(LIB_openldap)
 
 else
 
 ldap = hdb-ldap.c
+ldap_lib = $(LIB_openldap)
 
 endif
 
@@ -78,7 +80,7 @@
 	../krb5/libkrb5.la \
 	../asn1/libasn1.la \
 	$(LIBADD_roken) \
-	$(LIB_openldap) \
+	$(ldap_lib) \
 	$(LIB_dlopen) \
 	$(DBLIB) \
 	$(LIB_NDBM)
--- heimdal-0.8.1/kadmin/Makefile.am.orig	2007-04-23 18:24:45.000000000 +0200
+++ heimdal-0.8.1/kadmin/Makefile.am	2007-07-05 23:44:05.368653419 +0200
@@ -62,7 +62,6 @@
 
 LDADD_common = \
 	$(top_builddir)/lib/hdb/libhdb.la \
-	$(LIB_openldap) \
 	$(top_builddir)/lib/krb5/libkrb5.la \
 	$(LIB_des) \
 	$(top_builddir)/lib/asn1/libasn1.la \
--- heimdal-0.8.1/kcm/Makefile.am.orig	2007-04-23 18:24:31.000000000 +0200
+++ heimdal-0.8.1/kcm/Makefile.am	2007-07-05 23:43:51.151843250 +0200
@@ -32,7 +32,6 @@
 man_MANS = kcm.8
 
 LDADD = $(top_builddir)/lib/hdb/libhdb.la \
-	$(LIB_openldap) \
 	$(top_builddir)/lib/krb5/libkrb5.la \
 	$(LIB_krb4) \
 	$(LIB_des) \
--- heimdal-0.8.1/kdc/Makefile.am.orig	2007-04-23 18:24:42.000000000 +0200
+++ heimdal-0.8.1/kdc/Makefile.am	2007-07-05 23:43:29.950635061 +0200
@@ -60,7 +60,6 @@
 
 hprop_LDADD = \
 	$(top_builddir)/lib/hdb/libhdb.la \
-	$(LIB_openldap) \
 	$(top_builddir)/lib/krb5/libkrb5.la \
 	$(LIB_kdb) $(LIB_krb4) \
 	$(LIB_des) \
@@ -70,7 +69,6 @@
 
 hpropd_LDADD = \
 	$(top_builddir)/lib/hdb/libhdb.la \
-	$(LIB_openldap) \
 	$(top_builddir)/lib/krb5/libkrb5.la \
 	$(LIB_kdb) $(LIB_krb4) \
 	$(LIB_des) \
@@ -85,7 +83,6 @@
 libkdc_la_LIBADD = \
 	$(LIB_pkinit) \
 	$(top_builddir)/lib/hdb/libhdb.la \
-	$(LIB_openldap) \
 	$(top_builddir)/lib/krb5/libkrb5.la \
 	$(LIB_kdb) $(LIB_krb4) \
 	$(top_builddir)/lib/ntlm/libheimntlm.la \
@@ -95,7 +92,6 @@
 	$(DBLIB) 
 
 LDADD = $(top_builddir)/lib/hdb/libhdb.la \
-	$(LIB_openldap) \
 	$(top_builddir)/lib/krb5/libkrb5.la \
 	$(LIB_krb4) \
 	$(LIB_des) \
--- heimdal-0.8.1/kpasswd/Makefile.am.orig	2007-04-23 18:24:46.000000000 +0200
+++ heimdal-0.8.1/kpasswd/Makefile.am	2007-07-05 23:43:11.721596248 +0200
@@ -19,7 +19,6 @@
 kpasswdd_LDADD = \
 	$(top_builddir)/lib/kadm5/libkadm5srv.la \
 	$(top_builddir)/lib/hdb/libhdb.la \
-	$(LIB_openldap) \
 	$(LDADD) \
 	$(LIB_pidfile) \
 	$(LIB_dlopen) \
--- heimdal-0.8.1/lib/kadm5/Makefile.am.orig	2007-04-23 18:23:50.000000000 +0200
+++ heimdal-0.8.1/lib/kadm5/Makefile.am	2007-07-05 23:40:54.657785427 +0200
@@ -117,7 +117,6 @@
 LDADD = \
 	libkadm5srv.la \
 	$(top_builddir)/lib/hdb/libhdb.la \
-	$(LIB_openldap) \
 	$(top_builddir)/lib/krb5/libkrb5.la \
 	$(top_builddir)/lib/asn1/libasn1.la \
 	$(LIB_des) \
@@ -129,7 +128,6 @@
 iprop_log_LDADD = \
 	libkadm5srv.la \
 	$(top_builddir)/lib/hdb/libhdb.la \
-	$(LIB_openldap) \
 	$(top_builddir)/lib/krb5/libkrb5.la \
 	$(top_builddir)/lib/asn1/libasn1.la \
 	$(LIB_des) \
