--- heimdal-0.6.3/cf/krb-readline.m4.orig	2004-04-01 09:27:34.000000000 +0200
+++ heimdal-0.6.3/cf/krb-readline.m4	2005-03-10 21:49:10.738191736 +0100
@@ -22,17 +22,21 @@
 dnl readline
 
 ac_foo=no
+build_editline=no
 if test "$with_readline" = yes; then
 	:
 elif test "$ac_cv_func_readline" = yes; then
 	:
 elif test "$ac_cv_func_el_init" = yes; then
 	ac_foo=yes
+	build_editline=yes
 	LIB_readline="\$(top_builddir)/lib/editline/libel_compat.la \$(LIB_el_init) \$(LIB_tgetent)"
 else
+	build_editline=yes
 	LIB_readline="\$(top_builddir)/lib/editline/libeditline.la \$(LIB_tgetent)"
 fi
 AM_CONDITIONAL(el_compat, test "$ac_foo" = yes)
+AM_CONDITIONAL(BUILD_EDITLINE, test "$build_editline" = yes)
 AC_DEFINE(HAVE_READLINE, 1, 
 	[Define if you have a readline compatible library.])dnl
 
--- heimdal-1.0.1/lib/Makefile.am.orig	2007-08-09 09:47:21.000000000 +0200
+++ heimdal-1.0.1/lib/Makefile.am	2007-08-12 21:21:25.502963906 +0200
@@ -11,6 +11,9 @@
 if DCE
 dir_dce = kdfs
 endif
+if BUILD_EDITLINE
+dir_el = editline
+endif
 if COM_ERR
 dir_com_err = com_err
 endif
@@ -18,5 +21,5 @@
 dir_hcrypto = hcrypto
 endif
 
-SUBDIRS = roken vers editline $(dir_com_err) sl asn1 $(dir_hcrypto) hx509 \
+SUBDIRS = roken vers $(dir_el) $(dir_com_err) sl asn1 $(dir_hcrypto) hx509 \
 	krb5 ntlm kafs gssapi hdb kadm5 auth $(dir_45) $(dir_otp) $(dir_dce)
