--- heimdal-0.5.2/lib/hdb/hdb-ldap.c    2002-09-04 13:42:22.000000000 -0500
+++ heimdal/lib/hdb/hdb-ldap.c  2003-05-26 12:56:46.000000000 -0500
@@ -32,7 +32,7 @@

 #include "hdb_locl.h"

-RCSID("$Id$");
+RCSID("$Id$");

 #ifdef OPENLDAP

@@ -953,6 +953,12 @@
 static krb5_error_code LDAP__connect(krb5_context context, HDB * db)
 {
     int rc, version = LDAP_VERSION3;
+    /* Empty credentials to do a SASL bind with LDAP */
+    /* Note that empty different to NULL credentials */
+    /* If you provide NULL credentials instead of empty */
+    /* credentials you will get a SASL bind in progress message! */
+
+    struct berval bv = {0, ""};

     if (db->db != NULL) {
        /* connection has been opened. ping server. */
@@ -986,6 +992,14 @@
        return HDB_ERR_BADVERSION;
     }

+    rc = ldap_sasl_bind_s((LDAP *) db->db, NULL, "EXTERNAL", &bv, NULL, NULL,
NULL );
+    if (rc != LDAP_SUCCESS) {
+       krb5_set_error_string(context, "ldap_sasl_bind_s: %s", ldap_err2string(
rc));
+       ldap_unbind_ext((LDAP *) db->db, NULL, NULL);
+       db->db = NULL;
+       return HDB_ERR_BADVERSION;
+    }
+
     return 0;
 }

@@ -1104,7 +1118,7 @@
            ret = asprintf(&dn, "cn=%s,%s", name, db->name);
        } else {
            /* A bit bogus, but we don't have a search base */
-           ret = asprintf(&dn, "cn=%s", name, db->name);
+           ret = asprintf(&dn, "cn=%s", name);
        }
        if (ret < 0) {
            krb5_set_error_string(context, "asprintf: out of memory");
