--- heimdal-0.6.1/lib/roken/Makefile.am.orig	2004-04-02 20:09:47.000000000 +0200
+++ heimdal-0.6.1/lib/roken/Makefile.am	2004-04-02 20:40:54.064156064 +0200
@@ -92,7 +92,7 @@
 
 EXTRA_DIST = roken.awk roken.h.in $(EXTRA_libroken_SOURCES)
 
-libroken_la_LIBADD = @LTLIBOBJS@ $(DBLIB)
+libroken_la_LIBADD = @LTLIBOBJS@ $(DBLIB) $(LIB_crypt)
 
 $(LTLIBOBJS) $(libroken_la_OBJECTS): roken.h $(XHEADERS)
 
