--- heimdal-0.6/lib/roken/Makefile.am.orig	2003-07-26 03:39:57.000000000 +0200
+++ heimdal-0.6/lib/roken/Makefile.am	2003-07-26 09:48:39.000000000 +0200
@@ -84,7 +84,7 @@
 	write_pid.c		\
 	xdbm.h
 
-EXTRA_libroken_la_SOURCES =	\
+EXTRA_libroken_SOURCES =	\
 	chown.c			\
 	copyhostent.c		\
 	daemon.c		\
@@ -160,7 +160,7 @@
 	warnx.c			\
 	writev.c
 
-EXTRA_DIST = roken.awk roken.h.in
+EXTRA_DIST = roken.awk roken.h.in $(EXTRA_libroken_SOURCES)
 
 libroken_la_LIBADD = @LTLIBOBJS@ $(DBLIB)
 
