From 329918bd671c89de6e1c2874baba48d658a89a10 Mon Sep 17 00:00:00 2001
From: Damir Franusic <df@release14.org>
Date: Sun, 9 Dec 2018 19:53:58 +0100
Subject: [PATCH] hcrypto: fix include path

---
 lib/hcrypto/Makefile.am | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/lib/hcrypto/Makefile.am b/lib/hcrypto/Makefile.am
index 469176b6c6..195117d174 100644
--- a/lib/hcrypto/Makefile.am
+++ b/lib/hcrypto/Makefile.am
@@ -9,7 +9,8 @@ AM_CPPFLAGS += $(INCLUDE_openssl_crypto)
 endif
 
 AM_CPPFLAGS += -I$(top_srcdir)/lib/hx509 \
-	       -I$(srcdir)/libtommath -DUSE_HCRYPTO_LTM=1
+	       -I$(srcdir)/libtommath -DUSE_HCRYPTO_LTM=1 \
+	       -I$(srcdir)/..
 
 lib_LTLIBRARIES = libhcrypto.la
 check_LTLIBRARIES = libhctest.la
