diff -Nru heimdal-0.3c/admin/srvconvert.c heimdal-0.3c.new/admin/srvconvert.c
--- heimdal-0.3c/admin/srvconvert.c	Wed Nov  1 00:13:14 2000
+++ heimdal-0.3c.new/admin/srvconvert.c	Tue Oct 31 23:44:35 2000
@@ -38,7 +38,7 @@
 /* convert a version 4 srvtab to a version 5 keytab */
 
 #ifndef KEYFILE
-#define KEYFILE "/etc/srvtab"
+#define KEYFILE "/etc/heimdal/srvtab"
 #endif
 
 static char *srvtab = KEYFILE;
diff -Nru heimdal-0.3c/admin/srvcreate.c heimdal-0.3c.new/admin/srvcreate.c
--- heimdal-0.3c/admin/srvcreate.c	Wed Nov  1 00:13:14 2000
+++ heimdal-0.3c.new/admin/srvcreate.c	Tue Oct 31 23:44:35 2000
@@ -38,7 +38,7 @@
 /* convert a version 5 keytab to a version 4 srvtab */
 
 #ifndef KEYFILE
-#define KEYFILE "/etc/srvtab"
+#define KEYFILE "/etc/heimdal/srvtab"
 #endif
 
 static char *srvtab = KEYFILE;
diff -Nru heimdal-0.3c/appl/ftp/ftpd/logwtmp.c heimdal-0.3c.new/appl/ftp/ftpd/logwtmp.c
--- heimdal-0.3c/appl/ftp/ftpd/logwtmp.c	Wed Nov  1 00:13:14 2000
+++ heimdal-0.3c.new/appl/ftp/ftpd/logwtmp.c	Tue Oct 31 23:44:35 2000
@@ -65,7 +65,7 @@
 #ifdef _PATH_WTMP
 #define WTMP_FILE _PATH_WTMP
 #else
-#define WTMP_FILE "/var/adm/wtmp"
+#define WTMP_FILE "/var/run/wtmpx"
 #endif
 #endif
 
diff -Nru heimdal-0.3c/appl/ftp/ftpd/pathnames.h heimdal-0.3c.new/appl/ftp/ftpd/pathnames.h
--- heimdal-0.3c/appl/ftp/ftpd/pathnames.h	Wed Nov  1 00:13:14 2000
+++ heimdal-0.3c.new/appl/ftp/ftpd/pathnames.h	Tue Oct 31 23:44:35 2000
@@ -49,9 +49,9 @@
 #define _PATH_BSHELL "/bin/sh"
 #endif
 
-#define	_PATH_FTPUSERS		"/etc/ftpusers"
-#define	_PATH_FTPCHROOT		"/etc/ftpchroot"
-#define	_PATH_FTPWELCOME	"/etc/ftpwelcome"
+#define	_PATH_FTPUSERS		"/etc/ftpd/ftpusers"
+#define	_PATH_FTPCHROOT		"/etc/ftpd/ftpchroot"
+#define	_PATH_FTPWELCOME	"/etc/ftpd/ftpwelcome"
 #define	_PATH_FTPLOGINMESG	"/etc/motd"
 
 #define _PATH_ISSUE		"/etc/issue"
diff -Nru heimdal-0.3c/kdc/hpropd.c heimdal-0.3c.new/kdc/hpropd.c
--- heimdal-0.3c/kdc/hpropd.c	Wed Nov  1 00:13:14 2000
+++ heimdal-0.3c.new/kdc/hpropd.c	Tue Oct 31 23:44:35 2000
@@ -229,7 +229,7 @@
 
 #ifdef KRB4
     if (v4dump && database == HDB_DEFAULT_DB)
-       database = "/var/kerberos/524_dump";
+       database = "/var/heimdal/524_dump";
 #endif /* KRB4 */
     
     if(help_flag)
diff -Nru heimdal-0.3c/kdc/kdc.8 heimdal-0.3c.new/kdc/kdc.8
--- heimdal-0.3c/kdc/kdc.8	Wed Nov  1 00:13:14 2000
+++ heimdal-0.3c.new/kdc/kdc.8	Tue Oct 31 23:44:35 2000
@@ -98,9 +98,9 @@
 The configuration file has the same syntax as the 
 .Pa krb5.conf
 file (you can actually put the configuration in
-.Pa /etc/krb5.conf ,
+.Pa /etc/heimdal/krb5.conf ,
 and then start the KDC with
-.Fl -config-file= Ns Ar /etc/krb5.conf ) .
+.Fl -config-file= Ns Ar /etc/heimdal/krb5.conf ) .
 All options should be in a section called
 .Dq kdc .
 All the command-line options can preferably be added in the
diff -Nru heimdal-0.3c/kuser/kinit.1 heimdal-0.3c.new/kuser/kinit.1
--- heimdal-0.3c/kuser/kinit.1	Wed Nov  1 00:13:14 2000
+++ heimdal-0.3c.new/kuser/kinit.1	Tue Oct 31 23:44:35 2000
@@ -114,7 +114,7 @@
 The same as
 .Fl -keytab ,
 but with the default keytab name (normally
-.Ar FILE:/etc/krb5.keytab ) .
+.Ar FILE:/etc/heimdal/krb5.keytab ) .
 .It Xo
 .Fl v Ns ,
 .Fl -validate
diff -Nru heimdal-0.3c/lib/krb5/config_file.c heimdal-0.3c.new/lib/krb5/config_file.c
--- heimdal-0.3c/lib/krb5/config_file.c	Wed Nov  1 00:13:15 2000
+++ heimdal-0.3c.new/lib/krb5/config_file.c	Tue Oct 31 23:44:35 2000
@@ -757,7 +757,7 @@
 {
     krb5_config_section *c;
 
-    printf ("%d\n", krb5_config_parse_file ("/etc/krb5.conf", &c));
+    printf ("%d\n", krb5_config_parse_file ("/etc/heimdal/krb5.conf", &c));
     print_config (stdout, c);
     printf ("[libdefaults]ticket_lifetime = %s\n",
 	    krb5_config_get_string (context, c,
diff -Nru heimdal-0.3c/lib/krb5/constants.c heimdal-0.3c.new/lib/krb5/constants.c
--- heimdal-0.3c/lib/krb5/constants.c	Wed Nov  1 00:13:15 2000
+++ heimdal-0.3c.new/lib/krb5/constants.c	Tue Oct 31 23:44:35 2000
@@ -35,5 +35,5 @@
 
 RCSID("$Id$");
 
-const char krb5_config_file[] = "/etc/krb5.conf";
+const char krb5_config_file[] = "/etc/heimdal/krb5.conf";
 const char krb5_defkeyname[] = KEYTAB_DEFAULT;
diff -Nru heimdal-0.3c/lib/krb5/krb5.conf.5 heimdal-0.3c.new/lib/krb5/krb5.conf.5
--- heimdal-0.3c/lib/krb5/krb5.conf.5	Wed Nov  1 00:13:15 2000
+++ heimdal-0.3c.new/lib/krb5/krb5.conf.5	Tue Oct 31 23:44:35 2000
@@ -4,7 +4,7 @@
 .Dt KRB5.CONF 5
 .Os HEIMDAL
 .Sh NAME
-.Nm /etc/krb5.conf
+.Nm /etc/heimdal/krb5.conf
 .Nd
 Configuration file for Kerberos 5
 .Sh DESCRIPTION
diff -Nru heimdal-0.3c/lib/krb5/krb5_locl.h heimdal-0.3c.new/lib/krb5/krb5_locl.h
--- heimdal-0.3c/lib/krb5/krb5_locl.h	Wed Nov  1 00:13:15 2000
+++ heimdal-0.3c.new/lib/krb5/krb5_locl.h	Tue Oct 31 23:44:35 2000
@@ -127,7 +127,7 @@
 #define ALLOC_SEQ(X, N) do { (X)->len = (N); ALLOC((X)->val, (N)); } while(0)
 
 /* should this be public? */
-#define KEYTAB_DEFAULT "FILE:/etc/krb5.keytab"
+#define KEYTAB_DEFAULT "FILE:/etc/heimdal/krb5.keytab"
 
 #ifndef O_BINARY
 #define O_BINARY 0
