--- heimdal-0.6/lib/krb5/verify_krb5_conf.c.orig	2003-03-29 10:52:50.000000000 +0100
+++ heimdal-0.6/lib/krb5/verify_krb5_conf.c	2003-07-26 10:02:37.000000000 +0200
@@ -449,8 +449,8 @@
 
 #if 0
 struct entry kdcdefaults_entries[] = {
-    { "kdc_ports, krb5_config_string, mit_entry },
-    { "v4_mode, krb5_config_string, mit_entry },
+    { "kdc_ports", krb5_config_string, mit_entry },
+    { "v4_mode", krb5_config_string, mit_entry },
     { NULL }
 };
 #endif
