diff -ur heimdal-1.3.1/appl/dceutils/Makefile.am heimdal-1.3.1-sbin/appl/dceutils/Makefile.am
--- heimdal-1.3.1/appl/dceutils/Makefile.am	2009-11-21 16:41:51.000000000 +0100
+++ heimdal-1.3.1-sbin/appl/dceutils/Makefile.am	2009-12-07 17:20:32.677628994 +0100
@@ -8,7 +8,7 @@
 AIX_DFSPROGS = dpagaix
 endif
 
-libexec_PROGRAMS = $(DFSPROGS) $(AIX_DFSPROGS)
+sbin_PROGRAMS = $(DFSPROGS) $(AIX_DFSPROGS)
 
 dpagaix_CFLAGS = $(dpagaix_cflags)
 dpagaix_LDFLAGS = $(dpagaix_ldflags)
diff -ur heimdal-1.3.1/appl/ftp/ftpd/Makefile.am heimdal-1.3.1-sbin/appl/ftp/ftpd/Makefile.am
--- heimdal-1.3.1/appl/ftp/ftpd/Makefile.am	2009-11-21 16:41:51.000000000 +0100
+++ heimdal-1.3.1-sbin/appl/ftp/ftpd/Makefile.am	2009-12-07 17:20:32.678628545 +0100
@@ -4,7 +4,7 @@
 
 AM_CPPFLAGS += -I$(srcdir)/../common $(INCLUDE_krb4) -DFTP_SERVER
 
-libexec_PROGRAMS = ftpd
+sbin_PROGRAMS = ftpd
 
 CHECK_LOCAL = 
 
diff -ur heimdal-1.3.1/appl/kf/Makefile.am heimdal-1.3.1-sbin/appl/kf/Makefile.am
--- heimdal-1.3.1/appl/kf/Makefile.am	2009-11-21 16:41:51.000000000 +0100
+++ heimdal-1.3.1-sbin/appl/kf/Makefile.am	2009-12-07 17:20:32.681638741 +0100
@@ -4,7 +4,7 @@
 
 bin_PROGRAMS = kf
 
-libexec_PROGRAMS = kfd
+sbin_PROGRAMS = kfd
 
 man_MANS = kf.1 kfd.8
 
diff -ur heimdal-1.3.1/appl/kx/Makefile.am heimdal-1.3.1-sbin/appl/kx/Makefile.am
--- heimdal-1.3.1/appl/kx/Makefile.am	2009-11-21 16:41:51.000000000 +0100
+++ heimdal-1.3.1-sbin/appl/kx/Makefile.am	2009-12-07 17:20:32.684638988 +0100
@@ -10,13 +10,13 @@
 
 bin_PROGRAMS = kx
 bin_SCRIPTS = rxterm rxtelnet tenletxr
-libexec_PROGRAMS = kxd
+sbin_PROGRAMS = kxd
 
 else
 
 bin_PROGRAMS = 
 bin_SCRIPTS = 
-libexec_PROGRAMS = 
+sbin_PROGRAMS = 
 
 endif
 
diff -ur heimdal-1.3.1/appl/popper/Makefile.am heimdal-1.3.1-sbin/appl/popper/Makefile.am
--- heimdal-1.3.1/appl/popper/Makefile.am	2009-11-21 16:41:51.000000000 +0100
+++ heimdal-1.3.1-sbin/appl/popper/Makefile.am	2009-12-07 17:20:32.687628025 +0100
@@ -4,7 +4,7 @@
 
 noinst_PROGRAMS = pop_debug
 
-libexec_PROGRAMS = popper
+sbin_PROGRAMS = popper
 
 popper_SOURCES =				\
 	pop_auth.c				\
diff -ur heimdal-1.3.1/appl/push/Makefile.am heimdal-1.3.1-sbin/appl/push/Makefile.am
--- heimdal-1.3.1/appl/push/Makefile.am	2009-11-21 16:41:51.000000000 +0100
+++ heimdal-1.3.1-sbin/appl/push/Makefile.am	2009-12-07 17:20:32.690628323 +0100
@@ -6,7 +6,7 @@
 
 bin_SCRIPTS		= pfrom
 
-libexec_PROGRAMS	= push
+sbin_PROGRAMS	= push
 
 push_SOURCES = push.c push_locl.h
 
diff -ur heimdal-1.3.1/appl/rsh/Makefile.am heimdal-1.3.1-sbin/appl/rsh/Makefile.am
--- heimdal-1.3.1/appl/rsh/Makefile.am	2009-11-21 16:41:51.000000000 +0100
+++ heimdal-1.3.1-sbin/appl/rsh/Makefile.am	2009-12-07 17:20:32.693503567 +0100
@@ -8,7 +8,7 @@
 
 man_MANS = rsh.1 rshd.8
 
-libexec_PROGRAMS = rshd
+sbin_PROGRAMS = rshd
 
 rsh_SOURCES  = rsh.c common.c rsh_locl.h
 
diff -ur heimdal-1.3.1/appl/telnet/telnetd/Makefile.am heimdal-1.3.1-sbin/appl/telnet/telnetd/Makefile.am
--- heimdal-1.3.1/appl/telnet/telnetd/Makefile.am	2009-11-21 16:41:51.000000000 +0100
+++ heimdal-1.3.1-sbin/appl/telnet/telnetd/Makefile.am	2009-12-07 17:20:32.696628158 +0100
@@ -4,7 +4,7 @@
 
 AM_CPPFLAGS += -I$(srcdir)/.. $(INCLUDE_hcrypto)
 
-libexec_PROGRAMS = telnetd
+sbin_PROGRAMS = telnetd
 
 CHECK_LOCAL = 
 
diff -ur heimdal-1.3.1/kadmin/Makefile.am heimdal-1.3.1-sbin/kadmin/Makefile.am
--- heimdal-1.3.1/kadmin/Makefile.am	2009-12-07 17:20:42.684504134 +0100
+++ heimdal-1.3.1-sbin/kadmin/Makefile.am	2009-12-07 17:20:57.712503501 +0100
@@ -4,9 +4,7 @@
 
 AM_CPPFLAGS += $(INCLUDE_libintl) $(INCLUDE_readline) $(INCLUDE_hcrypto) -I$(srcdir)/../lib/krb5 -I$(top_builddir)/include/gssapi
 
-sbin_PROGRAMS = kadmin
-
-libexec_PROGRAMS = kadmind
+sbin_PROGRAMS = kadmin kadmind
 
 man_MANS = kadmin.8 kadmind.8
 
diff -ur heimdal-1.3.1/kcm/Makefile.am heimdal-1.3.1-sbin/kcm/Makefile.am
--- heimdal-1.3.1/kcm/Makefile.am	2009-12-07 17:20:42.687503717 +0100
+++ heimdal-1.3.1-sbin/kcm/Makefile.am	2009-12-07 17:20:32.702508680 +0100
@@ -4,7 +4,7 @@
 
 AM_CPPFLAGS += $(INCLUDE_libintl) $(INCLUDE_krb4) $(INCLUDE_hcrypto) -I$(srcdir)/../lib/krb5
 
-libexec_PROGRAMS = kcm
+sbin_PROGRAMS = kcm
 
 kcm_SOURCES =		\
 	acl.c		\
diff -ur heimdal-1.3.1/kdc/Makefile.am heimdal-1.3.1-sbin/kdc/Makefile.am
--- heimdal-1.3.1/kdc/Makefile.am	2009-12-07 17:20:42.690503160 +0100
+++ heimdal-1.3.1-sbin/kdc/Makefile.am	2009-12-07 17:21:16.991503983 +0100
@@ -8,9 +8,7 @@
 
 bin_PROGRAMS = string2key
 
-sbin_PROGRAMS = kstash
-
-libexec_PROGRAMS = hprop hpropd kdc digest-service
+sbin_PROGRAMS = kstash hprop hpropd kdc digest-service
 
 noinst_PROGRAMS = kdc-replay
 
diff -ur heimdal-1.3.1/kpasswd/Makefile.am heimdal-1.3.1-sbin/kpasswd/Makefile.am
--- heimdal-1.3.1/kpasswd/Makefile.am	2009-12-07 17:20:42.693503140 +0100
+++ heimdal-1.3.1-sbin/kpasswd/Makefile.am	2009-12-07 17:20:32.708638409 +0100
@@ -10,7 +10,7 @@
 
 kpasswd_SOURCES = kpasswd.c kpasswd_locl.h
 
-libexec_PROGRAMS = kpasswdd
+sbin_PROGRAMS = kpasswdd
 
 noinst_PROGRAMS = kpasswd-generator
 
--- heimdal-1.6/kuser/Makefile.am~	2015-02-07 06:15:40.000000000 +0200
+++ heimdal-1.6/kuser/Makefile.am	2016-05-01 13:15:58.861442172 +0300
@@ -16,7 +16,7 @@
 	kimpersonate.8
 
 bin_PROGRAMS = kinit kdestroy kgetcred heimtools
-libexec_PROGRAMS = kdigest kimpersonate
+sbin_PROGRAMS = kdigest kimpersonate
 
 noinst_PROGRAMS = kverify kdecode_ticket generate-requests
 
diff -ur heimdal-1.3.1/lib/kadm5/Makefile.am heimdal-1.3.1-sbin/lib/kadm5/Makefile.am
--- heimdal-1.3.1/lib/kadm5/Makefile.am	2009-12-07 17:20:42.696503142 +0100
+++ heimdal-1.3.1-sbin/lib/kadm5/Makefile.am	2009-12-07 17:21:52.790508435 +0100
@@ -10,7 +10,7 @@
 libkadm5srv_la_LDFLAGS += $(LDFLAGS_VERSION_SCRIPT)$(srcdir)/version-script.map
 endif
 
-sbin_PROGRAMS = iprop-log
+sbin_PROGRAMS = iprop-log ipropd-master ipropd-slave
 check_PROGRAMS = default_keys
 noinst_PROGRAMS = test_pw_quality
 
@@ -25,8 +25,6 @@
 libkadm5clnt_la_LIBADD = \
 	$(LIB_com_err) ../krb5/libkrb5.la $(LIBADD_roken)
 
-libexec_PROGRAMS = ipropd-master ipropd-slave
-
 default_keys_SOURCES = default_keys.c
 
 kadm5includedir = $(includedir)/kadm5
