diff -urbB heimdal-0.7.org/lib/hdb/hdb.c heimdal-0.7/lib/hdb/hdb.c
--- heimdal-0.7.org/lib/hdb/hdb.c	2005-06-16 18:28:04.000000000 +0200
+++ heimdal-0.7/lib/hdb/hdb.c	2005-07-03 19:03:44.222977552 +0200
@@ -361,7 +361,7 @@
 
     if(filename == NULL)
 	filename = HDB_DEFAULT_DB;
-    krb5_add_et_list(context, initialize_hdb_error_table_r);
+    krb5_add_et_list(context, initialize_hdb_error_table);
     h = find_method (filename, &residual);
 #ifdef HAVE_DLOPEN
     if (h == NULL)
diff -urbB heimdal-0.7.org/lib/kadm5/context_s.c heimdal-0.7/lib/kadm5/context_s.c
--- heimdal-0.7.org/lib/kadm5/context_s.c	2005-06-16 18:28:07.000000000 +0200
+++ heimdal-0.7/lib/kadm5/context_s.c	2005-07-03 19:03:44.222977552 +0200
@@ -182,7 +182,7 @@
     memset(*ctx, 0, sizeof(**ctx));
     set_funcs(*ctx);
     (*ctx)->context = context;
-    krb5_add_et_list (context, initialize_kadm5_error_table_r);
+    krb5_add_et_list (context, initialize_kadm5_error_table);
 #define is_set(M) (params && params->mask & KADM5_CONFIG_ ## M)
     if(is_set(REALM))
 	(*ctx)->config.realm = strdup(params->realm);
diff -urbB heimdal-0.7.org/lib/kadm5/init_c.c heimdal-0.7/lib/kadm5/init_c.c
--- heimdal-0.7.org/lib/kadm5/init_c.c	2005-06-16 18:28:07.000000000 +0200
+++ heimdal-0.7/lib/kadm5/init_c.c	2005-07-03 19:03:44.223977400 +0200
@@ -69,7 +69,7 @@
     if(*ctx == NULL)
 	return ENOMEM;
     memset(*ctx, 0, sizeof(**ctx));
-    krb5_add_et_list (context, initialize_kadm5_error_table_r);
+    krb5_add_et_list (context, initialize_kadm5_error_table);
     set_funcs(*ctx);
     (*ctx)->context = context;
     if(params->mask & KADM5_CONFIG_REALM) {
diff -urbB heimdal-0.7.org/lib/kadm5/kadm5_err.et heimdal-0.7/lib/kadm5/kadm5_err.et
--- heimdal-0.7.org/lib/kadm5/kadm5_err.et	2005-06-16 18:28:07.000000000 +0200
+++ heimdal-0.7/lib/kadm5/kadm5_err.et	2005-07-03 19:03:44.219978008 +0200
@@ -45,10 +45,10 @@
 error_code NEW_STRUCT_VERSION,	"API structure version specified by application is unknown to libraries"
 error_code BAD_API_VERSION,	"Programmer error!  Bad API version"
 error_code OLD_LIB_API_VERSION,	"API version specified by application is no longer supported by libraries"
-error_code OLD_SERVER_API_VERSION,"API version specified by application is no longer supported by server"
+error_code OLD_SERVER_API_VERSION, "API version specified by application is no longer supported by server"
 error_code NEW_LIB_API_VERSION,	"API version specified by application is unknown to libraries"
-error_code NEW_SERVER_API_VERSION,"API version specified by application is unknown to server"
-error_code SECURE_PRINC_MISSING,"Database error! Required principal missing"
+error_code NEW_SERVER_API_VERSION, "API version specified by application is unknown to server"
+error_code SECURE_PRINC_MISSING, "Database error! Required principal missing"
 error_code NO_RENAME_SALT,	"The salt type of the specified principal does not support renaming"
 error_code BAD_CLIENT_PARAMS,	"Invalid configuration parameter for remote KADM5 client"
 error_code BAD_SERVER_PARAMS,	"Invalid configuration parameter for local KADM5 client."
diff -urbB heimdal-0.7.org/lib/krb5/context.c heimdal-0.7/lib/krb5/context.c
--- heimdal-0.7.org/lib/krb5/context.c	2005-06-16 18:28:08.000000000 +0200
+++ heimdal-0.7/lib/krb5/context.c	2005-07-03 19:03:44.221977704 +0200
@@ -531,10 +531,10 @@
 krb5_init_ets(krb5_context context)
 {
     if(context->et_list == NULL){
-	krb5_add_et_list(context, initialize_krb5_error_table_r);
-	krb5_add_et_list(context, initialize_asn1_error_table_r);
-	krb5_add_et_list(context, initialize_heim_error_table_r);
-	krb5_add_et_list(context, initialize_k524_error_table_r);
+	krb5_add_et_list(context, initialize_krb5_error_table);
+	krb5_add_et_list(context, initialize_asn1_error_table);
+	krb5_add_et_list(context, initialize_heim_error_table);
+	krb5_add_et_list(context, initialize_k524_error_table);
     }
 }
 
diff -urbB heimdal-0.7.org/lib/krb5/krb5_err.et heimdal-0.7/lib/krb5/krb5_err.et
--- heimdal-0.7.org/lib/krb5/krb5_err.et	2005-06-16 18:28:09.000000000 +0200
+++ heimdal-0.7/lib/krb5/krb5_err.et	2005-07-03 19:03:44.220977856 +0200
@@ -16,7 +16,7 @@
 error_code S_OLD_MAST_KVNO,	"Server's key is encrypted in an old master key"
 error_code C_PRINCIPAL_UNKNOWN,	"Client not found in Kerberos database"
 error_code S_PRINCIPAL_UNKNOWN,	"Server not found in Kerberos database"
-error_code PRINCIPAL_NOT_UNIQUE,"Principal has multiple entries in Kerberos database"
+error_code PRINCIPAL_NOT_UNIQUE, "Principal has multiple entries in Kerberos database"
 error_code NULL_KEY,		"Client or server has a null key"
 error_code CANNOT_POSTDATE,	"Ticket is ineligible for postdating"
 error_code NEVER_VALID,		"Requested effective lifetime is negative or too short"
