commit 826c14833f98d1c52830361f4b5a76e3344f388c
Author: Love Hornquist Astrand <lha@h5l.org>
Date:   Mon Nov 23 07:19:00 2009 -0800

    make cpp safe, from Allan McRae

diff --git a/lib/krb5/cache.c b/lib/krb5/cache.c
index 505ecd0..42e1877 100644
--- a/lib/krb5/cache.c
+++ b/lib/krb5/cache.c
@@ -1376,7 +1376,7 @@ out:
  *
  */
 
-struct krb5_cccol_cursor {
+struct krb5_cccol_cursor_data {
     int idx;
     krb5_cc_cache_cursor cursor;
 };
diff --git a/lib/krb5/krb5.h b/lib/krb5/krb5.h
index 9d2ecb3..8fda340 100644
--- a/lib/krb5/krb5.h
+++ b/lib/krb5/krb5.h
@@ -313,7 +313,7 @@ struct krb5_cc_ops;
 				 NULL)
 
 typedef void *krb5_cc_cursor;
-typedef struct krb5_cccol_cursor *krb5_cccol_cursor;
+typedef struct krb5_cccol_cursor_data *krb5_cccol_cursor;
 
 typedef struct krb5_ccache_data {
     const struct krb5_cc_ops *ops;
