--- heimdal-1.3.1/lib/gssapi/ntlm/accept_sec_context.c.orig	2009-11-21 16:41:51.000000000 +0100
+++ heimdal-1.3.1/lib/gssapi/ntlm/accept_sec_context.c	2009-12-09 12:36:21.000000000 +0100
@@ -44,7 +44,24 @@
     struct ntlm_server_interface *ns_interface = NULL;
     
 #ifdef DIGEST
+    krb5_context context = NULL;
+    krb5_error_code ret;
+    krb5_boolean b;
+
+    ret = krb5_init_context(&context);
+    if (ret)
+	return GSS_S_FAILURE;
+
+    b = krb5_config_get_bool_default(context, NULL, FALSE,
+				"gssapi",
+				"disable-ntlm-digest", NULL);
+
+    krb5_free_context(context);
+
+    if (b)
+	return GSS_S_FAILURE;
+
     ns_interface = &ntlmsspi_kdc_digest;
 #endif
     if (ns_interface == NULL)
 	return GSS_S_FAILURE;
