--- heimdal-1.2.1/lib/gssapi/ntlm/accept_sec_context.c.orig	2008-07-28 14:04:39.000000000 +0200
+++ heimdal-1.2.1/lib/gssapi/ntlm/accept_sec_context.c	2009-12-09 21:40:30.074974606 +0100
@@ -43,6 +43,22 @@
 _gss_ntlm_allocate_ctx(OM_uint32 *minor_status, ntlm_ctx *ctx)
 {
     OM_uint32 maj_stat;
+    krb5_context context = NULL;
+    krb5_error_code ret;
+    krb5_boolean b;
+
+    ret = krb5_init_context(&context);
+    if (ret)
+	return GSS_S_FAILURE;
+
+    b = krb5_config_get_bool_default(context, NULL, FALSE,
+				"gssapi",
+				"disable-ntlm-digest", NULL);
+
+    krb5_free_context(context);
+
+    if (b)
+	return GSS_S_FAILURE;
 
     *ctx = calloc(1, sizeof(**ctx));
 
