diff -urNp -x '*.orig' heimdal-7.7.0.org/lib/roken/roken-common.h heimdal-7.7.0/lib/roken/roken-common.h
--- heimdal-7.7.0.org/lib/roken/roken-common.h	2019-06-07 08:21:35.000000000 +0200
+++ heimdal-7.7.0/lib/roken/roken-common.h	2021-05-10 00:32:28.783798762 +0200
@@ -294,7 +294,8 @@ ROKEN_CPP_START
 #ifndef _WIN32
 #ifdef RETSIGTYPE
 typedef RETSIGTYPE (*SigAction)(int);
-SigAction signal(int iSig, SigAction pAction); /* BSD compatible */
+#define signal rk_signal
+SigAction ROKEN_LIB_FUNCTION signal(int iSig, SigAction pAction); /* BSD compatible */
 #endif
 #endif
 #endif
diff -urNp -x '*.orig' heimdal-7.7.0.org/lib/roken/version-script.map heimdal-7.7.0/lib/roken/version-script.map
--- heimdal-7.7.0.org/lib/roken/version-script.map	2019-06-07 08:21:35.000000000 +0200
+++ heimdal-7.7.0/lib/roken/version-script.map	2021-05-10 00:32:28.784798762 +0200
@@ -95,6 +95,7 @@ HEIMDAL_ROKEN_1.0 {
 		rk_strerror;
 		rk_strerror_r;
 		rk_setprogname;
+		rk_signal;
 		rk_simple_execle;
 		rk_simple_execlp;
 		rk_simple_execve;
@@ -183,7 +184,6 @@ HEIMDAL_ROKEN_1.0 {
 		rtbl_set_flags;
 		rtbl_set_prefix;
 		rtbl_set_separator;
-		signal;
 		simple_execl;
 		tm2time;
 		unix_verify_user;
diff -urNp -x '*.orig' heimdal-7.7.0.org/lib/sl/sl.c heimdal-7.7.0/lib/sl/sl.c
--- heimdal-7.7.0.org/lib/sl/sl.c	2016-12-20 15:23:06.000000000 +0100
+++ heimdal-7.7.0/lib/sl/sl.c	2021-05-10 00:32:28.784798762 +0200
@@ -36,6 +36,11 @@
 #include "sl_locl.h"
 #include <setjmp.h>
 
+/* take it from glibc (libsl and libss are not linked with libroken) */
+#ifdef signal
+#undef signal
+#endif
+
 static void
 mandoc_template(SL_cmd *cmds,
 		const char *extra_string)
