diff -urNp -x '*.orig' heimdal-7.7.0.org/appl/dceutils/Makefile.am heimdal-7.7.0/appl/dceutils/Makefile.am
--- heimdal-7.7.0.org/appl/dceutils/Makefile.am	2016-12-20 15:23:06.000000000 +0100
+++ heimdal-7.7.0/appl/dceutils/Makefile.am	2021-05-10 00:32:51.411798762 +0200
@@ -8,7 +8,7 @@ if AIX
 AIX_DFSPROGS = dpagaix
 endif
 
-libexec_PROGRAMS = $(DFSPROGS) $(AIX_DFSPROGS)
+sbin_PROGRAMS = $(DFSPROGS) $(AIX_DFSPROGS)
 
 dpagaix_CFLAGS = $(dpagaix_cflags)
 dpagaix_LDFLAGS = $(dpagaix_ldflags)
diff -urNp -x '*.orig' heimdal-7.7.0.org/appl/kf/Makefile.am heimdal-7.7.0/appl/kf/Makefile.am
--- heimdal-7.7.0.org/appl/kf/Makefile.am	2016-12-20 15:23:06.000000000 +0100
+++ heimdal-7.7.0/appl/kf/Makefile.am	2021-05-10 00:32:51.411798762 +0200
@@ -4,7 +4,7 @@ include $(top_srcdir)/Makefile.am.common
 
 bin_PROGRAMS = kf
 
-libexec_PROGRAMS = kfd
+sbin_PROGRAMS = kfd
 
 man_MANS = kf.1 kfd.8
 
diff -urNp -x '*.orig' heimdal-7.7.0.org/kadmin/Makefile.am heimdal-7.7.0/kadmin/Makefile.am
--- heimdal-7.7.0.org/kadmin/Makefile.am	2019-06-07 08:21:35.000000000 +0200
+++ heimdal-7.7.0/kadmin/Makefile.am	2021-05-10 00:32:51.411798762 +0200
@@ -6,7 +6,7 @@ AM_CPPFLAGS += $(INCLUDE_libintl) $(INCL
 
 bin_PROGRAMS = kadmin
 
-libexec_PROGRAMS = kadmind
+sbin_PROGRAMS = kadmind
 
 man_MANS = kadmin.1 kadmind.8
 
diff -urNp -x '*.orig' heimdal-7.7.0.org/kcm/Makefile.am heimdal-7.7.0/kcm/Makefile.am
--- heimdal-7.7.0.org/kcm/Makefile.am	2016-12-20 15:23:06.000000000 +0100
+++ heimdal-7.7.0/kcm/Makefile.am	2021-05-10 00:32:51.411798762 +0200
@@ -4,7 +4,7 @@ include $(top_srcdir)/Makefile.am.common
 
 AM_CPPFLAGS += $(INCLUDE_libintl) -I$(srcdir)/../lib/krb5
 
-libexec_PROGRAMS = kcm
+sbin_PROGRAMS = kcm
 
 kcm_SOURCES =		\
 	acl.c		\
diff -urNp -x '*.orig' heimdal-7.7.0.org/kdc/Makefile.am heimdal-7.7.0/kdc/Makefile.am
--- heimdal-7.7.0.org/kdc/Makefile.am	2016-12-20 15:23:06.000000000 +0100
+++ heimdal-7.7.0/kdc/Makefile.am	2021-05-10 00:32:51.412798762 +0200
@@ -8,9 +8,7 @@ lib_LTLIBRARIES = libkdc.la
 
 bin_PROGRAMS = string2key
 
-sbin_PROGRAMS = kstash
-
-libexec_PROGRAMS = hprop hpropd kdc digest-service
+sbin_PROGRAMS = kstash hprop hpropd kdc digest-service
 
 noinst_PROGRAMS = kdc-replay kdc-tester
 
diff -urNp -x '*.orig' heimdal-7.7.0.org/kpasswd/Makefile.am heimdal-7.7.0/kpasswd/Makefile.am
--- heimdal-7.7.0.org/kpasswd/Makefile.am	2016-12-20 15:23:06.000000000 +0100
+++ heimdal-7.7.0/kpasswd/Makefile.am	2021-05-10 00:32:51.412798762 +0200
@@ -8,7 +8,7 @@ bin_PROGRAMS = kpasswd
 
 kpasswd_SOURCES = kpasswd.c kpasswd_locl.h
 
-libexec_PROGRAMS = kpasswdd
+sbin_PROGRAMS = kpasswdd
 
 noinst_PROGRAMS = kpasswd-generator
 
diff -urNp -x '*.orig' heimdal-7.7.0.org/kuser/Makefile.am heimdal-7.7.0/kuser/Makefile.am
--- heimdal-7.7.0.org/kuser/Makefile.am	2016-12-20 15:23:06.000000000 +0100
+++ heimdal-7.7.0/kuser/Makefile.am	2021-05-10 00:32:51.412798762 +0200
@@ -20,7 +20,7 @@ man_MANS = \
 	kimpersonate.8
 
 bin_PROGRAMS = kinit kdestroy kgetcred heimtools
-libexec_PROGRAMS = kdigest kimpersonate
+sbin_PROGRAMS = kdigest kimpersonate
 
 noinst_PROGRAMS = kverify kdecode_ticket generate-requests
 
diff -urNp -x '*.orig' heimdal-7.7.0.org/lib/kadm5/Makefile.am heimdal-7.7.0/lib/kadm5/Makefile.am
--- heimdal-7.7.0.org/lib/kadm5/Makefile.am	2019-06-07 08:21:35.000000000 +0200
+++ heimdal-7.7.0/lib/kadm5/Makefile.am	2021-05-10 00:32:51.412798762 +0200
@@ -13,7 +13,7 @@ libkadm5clnt_la_LDFLAGS += $(LDFLAGS_VER
 libkadm5srv_la_LDFLAGS += $(LDFLAGS_VERSION_SCRIPT)$(srcdir)/version-script.map
 endif
 
-sbin_PROGRAMS = iprop-log
+sbin_PROGRAMS = iprop-log ipropd-master ipropd-slave
 check_PROGRAMS = default_keys
 noinst_PROGRAMS = test_pw_quality
 
@@ -28,8 +28,6 @@ libkadm5srv_la_LIBADD = \
 libkadm5clnt_la_LIBADD = \
 	$(LIB_com_err) ../krb5/libkrb5.la $(LIBADD_roken)
 
-libexec_PROGRAMS = ipropd-master ipropd-slave
-
 default_keys_SOURCES = default_keys.c
 
 kadm5includedir = $(includedir)/kadm5
