diff -urNp -x '*.orig' heimdal-7.7.0.org/lib/roken/Makefile.am heimdal-7.7.0/lib/roken/Makefile.am
--- heimdal-7.7.0.org/lib/roken/Makefile.am	2019-06-07 08:21:35.000000000 +0200
+++ heimdal-7.7.0/lib/roken/Makefile.am	2021-05-10 00:32:07.632798762 +0200
@@ -142,6 +142,8 @@ EXTRA_libroken_la_SOURCES =	\
 
 libroken_la_LIBADD = @LTLIBOBJS@ $(LIB_crypt) $(LIB_pidfile)
 
+libroken_la_LIBADD += $(LIB_crypt)
+
 $(LTLIBOBJS) $(libroken_la_OBJECTS): roken.h $(XHEADERS)
 
 BUILT_SOURCES = roken.h
