--- heimdal-1.0.1/lib/roken/Makefile.am.orig	2007-08-09 09:47:19.000000000 +0200
+++ heimdal-1.0.1/lib/roken/Makefile.am	2007-08-12 21:18:46.005874686 +0200
@@ -101,7 +101,7 @@
 	write_pid.c		\
 	xdbm.h
 
-EXTRA_libroken_la_SOURCES =	\
+EXTRA_libroken_SOURCES =	\
 	err.hin			\
 	glob.hin		\
 	ifaddrs.hin		\
@@ -189,4 +189,5 @@
 	test-mem.h \
 	ndbm_wrap.c \
 	ndbm_wrap.h \
-	version-script.map
+	version-script.map \
+	$(EXTRA_libroken_SOURCES)
